From e7ea22488c8813530bee8ccd293195858dff1eff Mon Sep 17 00:00:00 2001
From: Yassin <yassin@users.noreply.github.com>
Date: Sat, 29 Aug 2026 19:55:52 +0200
Subject: [PATCH 1/6] dt-bindings: ufs: samsung: Add Exynos990 support

Add compatible strings for the Exynos990 UFS host, UFS PHY and HSI1
system registers so the storage controller can be described in the SoC
device tree.

Signed-off-by: Yassin <yassin@users.noreply.github.com>
---
 Documentation/devicetree/bindings/phy/samsung,ufs-phy.yaml       | 1 +
 .../devicetree/bindings/soc/samsung/samsung,exynos-sysreg.yaml   | 1 +
 Documentation/devicetree/bindings/ufs/samsung,exynos-ufs.yaml    | 1 +
 3 files changed, 3 insertions(+)

diff --git a/Documentation/devicetree/bindings/phy/samsung,ufs-phy.yaml b/Documentation/devicetree/bindings/phy/samsung,ufs-phy.yaml
index 2b20c0a5e509..e61686daa722 100644
--- a/Documentation/devicetree/bindings/phy/samsung,ufs-phy.yaml
+++ b/Documentation/devicetree/bindings/phy/samsung,ufs-phy.yaml
@@ -17,6 +17,7 @@ properties:
     enum:
       - google,gs101-ufs-phy
       - samsung,exynos7-ufs-phy
+      - samsung,exynos990-ufs-phy
       - samsung,exynosautov9-ufs-phy
       - samsung,exynosautov920-ufs-phy
       - tesla,fsd-ufs-phy
diff --git a/Documentation/devicetree/bindings/soc/samsung/samsung,exynos-sysreg.yaml b/Documentation/devicetree/bindings/soc/samsung/samsung,exynos-sysreg.yaml
index fcdb9ffd2644..19eeb283bc9a 100644
--- a/Documentation/devicetree/bindings/soc/samsung/samsung,exynos-sysreg.yaml
+++ b/Documentation/devicetree/bindings/soc/samsung/samsung,exynos-sysreg.yaml
@@ -35,6 +35,7 @@ properties:
               - samsung,exynos8895-fsys1-sysreg
               - samsung,exynos8895-peric0-sysreg
               - samsung,exynos8895-peric1-sysreg
+              - samsung,exynos990-hsi1-sysreg
               - samsung,exynos990-peric0-sysreg
               - samsung,exynos990-peric1-sysreg
               - samsung,exynosautov920-hsi0-sysreg
diff --git a/Documentation/devicetree/bindings/ufs/samsung,exynos-ufs.yaml b/Documentation/devicetree/bindings/ufs/samsung,exynos-ufs.yaml
index 710ce493f3b6..4132a1cdb537 100644
--- a/Documentation/devicetree/bindings/ufs/samsung,exynos-ufs.yaml
+++ b/Documentation/devicetree/bindings/ufs/samsung,exynos-ufs.yaml
@@ -17,6 +17,7 @@ properties:
     enum:
       - google,gs101-ufs
       - samsung,exynos7-ufs
+      - samsung,exynos990-ufs
       - samsung,exynosautov9-ufs
       - samsung,exynosautov9-ufs-vh
       - samsung,exynosautov920-ufs
